### PR TITLE
Azure Monitor: Add info about multiple resource selection

### LINKED ISSUE #62299: Add info notice about selecting multiple resources

### PR DESCRIPTION
**Pull request by andresmgot**
**What is this feature?**

Adds a info box with details about selection of multiple resources:

Screenshot from 2023-01-27 14-35-19

Depending on the selection, the message will change. 

**Which issue(s) does this PR fix?**:

Fixes #62299

**Special notes for your reviewer**:

I chose not to display any message when selecting a metric resource that doesn't have support for multiple selection because I found it too noisy but the Azure Portal does (even though the case is different since there you can upvote to add support for it). Happy to add also text in this case if you think it's helpful.

Screenshot from 2023-01-27 14-44-37

### CODE REVIEW

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/components/LogsQueryEditor/LogsQueryEditor.tsx [hunk 1]
```diff
@@ -82,6 +82,11 @@ const LogsQueryEditor: React.FC<LogsQueryEditorProps> = ({
                 // eslint-disable-next-line
                 <AdvancedResourcePicker resources={resources as string[]} onChange={onChange} />
               )}
+              selectionNotice={() =>
+                config.featureToggles.azureMultipleResourcePicker
+                  ? 'You may only choose items from the same resource type.'
```
Comment (aangelisc) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/components/LogsQueryEditor/LogsQueryEditor.tsx:87]:
```suggestion
                  ? 'You may only choose items of the same resource type.'
```

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/components/LogsQueryEditor/LogsQueryEditor.test.tsx [hunk 2]
```diff
@@ -118,4 +118,37 @@ describe('LogsQueryEdiutor', () => {
...
+    subscriptionButton.click();
+
+    const resourceGroupButton = await screen.findByRole('button', { name: 'Expand A Great Resource Group' });
+    resourceGroupButton.click();
+
+    const checkbox = await screen.findByLabelText('web-server');
+    await userEvent.click(checkbox);
+    expect(checkbox).toBeChecked();
+
+    expect(await screen.findByText('You may only choose items from the same resource type.')).toBeInTheDocument();
```
Comment (aangelisc) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/components/LogsQueryEditor/LogsQueryEditor.test.tsx:152]:
```suggestion
    expect(await screen.findByText('You may only choose items of the same resource type.')).toBeInTheDocument();
```

**Review by aangelisc — APPROVED** [review]
LGTM 👍

**Review by asimpson — APPROVED** [review]
Nice work!

**Pull request merged**